### PR TITLE
Add split to new dataset API

### PR DESCRIPTION
**Pull request by yueyericardo**

### CODE REVIEW

File: torchani/data/new.py [hunk 1]
```diff
@@ -273,14 +313,27 @@ def ShuffledDataset(file_path,
...
+                                                    shuffle=shuffle,
+                                                    num_workers=num_workers,
+                                                    pin_memory=False,
+                                                    collate_fn=my_collate_fn)
+    if val_size == 0:
+        return train_data_loader
+
+    val_data_loader = torch.utils.data.DataLoader(dataset=val_dataset,
+                                                  batch_size=batch_size,
+                                                  shuffle=shuffle,
```
Comment (farhadrgh) [inline torchani/data/new.py]: Do we need to shuffle the validation set?

Comment (yueyericardo) [reply torchani/data/new.py]: Theoretically, there should be no difference. But no shuffle might take less computation, I will change it to `False`
